### DISCUSSION TITLE
Get collision callbacks for all bodies

### DISCUSSION THREAD

**Discussion by bobbydigitales** [jrouwe/JoltPhysics#409] [Q&A] [answered]
Hey! I want to get ContactListener callbacks for all bodies added to my simulation. It seems from prior answers that the recommended way to do that is to add another body for each body and set that additional body to be a sensor. Is that the most efficient way? Is there any way to do this without doubling the number of bodies?

**Comment by jrouwe** [comment]
Can you please describe what you're trying to achieve? What do you mean with 'all bodies', also static vs static?

**Reply by bobbydigitales** [reply]
Ah sorry, only dynamic vs dynamic and dynamic vs static. It seemed that you need to make a sensor to get those callbacks? Or can you get callbacks for any body?

**Reply by jrouwe** [reply, marked as answer]
The ContactListener gives you callbacks for all those collisions, no sensors needed.
